src/main/java/uk/gov/companieshouse/api/accounts/validation/DebtorsValidator.java

```diff
@@ -183,6 +209,141 @@ private void validateInconsistentPeriodFiling(Debtors debtors, Errors errors) {
             addInconsistentDataError(errors, PREVIOUS_TOTAL_PATH);
         }
     }
-}
 
+    /**
+     * @inheritDoc
+     */
+
+    @Override
+    public Errors crossValidate(Errors errors, HttpServletRequest request, String CompanyAccountsId,
+                                Debtors debtors) throws DataException {
+
+        crossValidateCurrentPeriod(errors, request, debtors, CompanyAccountsId);
+        crossValidatePreviousPeriod(errors, request, CompanyAccountsId, debtors);
+
+        return errors;
+    }
+
+    private void crossValidatePreviousPeriod(Errors errors, HttpServletRequest request,
+                                             String CompanyAccountsId,
+                                             Debtors debtors) throws DataException {
+
+        String previousPeriodId = previousPeriodService.generateID(CompanyAccountsId);
+
+        // get previous period object
+        ResponseObject<PreviousPeriod> previousPeriodResponseObject;
+
+        try {
+            previousPeriodResponseObject =
+                previousPeriodService.findById(previousPeriodId, request);
+        } catch (MongoException e) {
+
+            throw new DataException(e.getMessage(), e);
+        }
```

Comment:
We could probably broke down this method into smaller, modular, reusable methods.
e.g. 

Line 231-242, it could be moved to its own method. 
```
  private ResponseObject<PreviousPeriod> getPreviousPeriodResponseObject(String companyAccountsId,
        HttpServletRequest request) throws DataException{
        String previousPeriodId = previousPeriodService.generateID(companyAccountsId);

        try {
            return previousPeriodService.findById(previousPeriodId, request);
        } catch (MongoException e) {

            throw new DataException(e.getMessage(), e);
        }
    }
```

So that we can change all that for: 
```
ResponseObject<PreviousPeriod> previousPeriodResponseObject = getPreviousPeriodResponseObject(
            CompanyAccountsId, request);
  ````

Reply:
new method created to getPreviousPeriodResponseObject in b39ab32